_posts/2020-11-09-pulling-most-recent-files.markdown

```diff
@@ -0,0 +1,78 @@
+---
+layout: post
+title:  "Pull most recent files in R"
+date:   2020-11-09
+categories: R base package
+---
+# Pulling most recent files in R
+ ###### by Joel Christian Andrade
+
+Many coding pipelines involve taking raw data files, formulating code and eventually working with a few, if not many, output files.
+
+Take these files in the directory `/raw_data` below, downloaded from publicly available Apple mobility database tracking movement.
+
+```
+applemobilitytrends_2020-08-29.csv
+applemobilitytrends_2020-09-12.csv
+applemobilitytrends_2020-09-18.csv
+applemobilitytrends_2020-09-25.csv
+applemobilitytrends_2020-09-30.csv
+applemobilitytrends_2020-10-02.csv
+applemobilitytrends_2020-11-08.csv
+```
+
+While retroactive data analysis is useful, the most recent file will include all data from previous files.  Thus, to retrieve the most up to date information while using R, we must use some functions which are pre-loaded into R.
+
+First, we must elucidate these functions
+* `file.info` = provides some basic information about a file(s)
+ * size
+ * if the file is directory
+ * if the file is an executable
+ * editing times
+   * `mtime` = modification time
+   * `atime` = access time
+   * `ctime` = creation time
+
+
+* `list.files` = list of all files within a directory
+
+* `which.min` or `which.max` = determine minimum or maximum with respect to a given set of variables
+
+Let's take the directory `raw_data` and view various outputs using these functions
+
+`file.info()`
+```r
+> file.info("/raw_data")
+
+              size  isdir  mode              mtime                ctime                atime  exe  
+data/raw_data    0   TRUE   777 2020-10-4 13:03:40  2020-09-19 17:14:40  2020-10-04 13:03:40  no
+```
+
+`list.files()`
+```r
+> list.files("/raw_data")
+[1] "applemobilitytrends_2020-09-12.csv" "applemobilitytrends_2020-09-18.csv" "applemobilitytrends_2020-09-25.csv"
+[4] "applemobilitytrends_2020-09-30.csv" "applemobilitytrends_2020-10-02.csv" "applemobilitytrends_2020-08-29.csv"
+```
```

Comment:
@jsandrad maybe it would be easier on the reader if these files matched the ones above?